core/drivers/crypto/crypto_api/include/drvcrypt_cipher.h

```diff
@@ -0,0 +1,73 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright 2018-2020 NXP
+ *
+ * Cipher interface calling the HW crypto driver.
+ */
+#ifndef __DRVCRYPT_CIPHER_H__
+#define __DRVCRYPT_CIPHER_H__
+
+#include <crypto/crypto_impl.h>
+#include <tee_api_types.h>
+
+/*
+ * Format the CIPHER context to keep the reference to the
+ * operation driver
+ */
+struct crypto_cipher {
+	struct crypto_cipher_ctx cipher_ctx; /* Crypto Cipher API context */
+	void *ctx;                           /* Cipher Context */
+	struct drvcrypt_cipher *op;          /* Reference to the operation */
+};
+
+/*
+ * Cipher Algorithm initialization data
+ */
+struct drvcrypt_cipher_init {
+	void *ctx;		  /* Software Context */
+	bool encrypt;		  /* Encrypt or decrypt direction */
+	struct drvcrypt_buf key1; /* First Key */
+	struct drvcrypt_buf key2; /* Second Key */
+	struct drvcrypt_buf iv;	  /* Initial Vector */
+};
+
+/*
+ * Cipher Algorithm update data
+ */
+struct drvcrypt_cipher_update {
+	void *ctx;		 /* Software Context */
+	bool encrypt;		 /* Encrypt or decrypt direction */
+	bool last;		 /* Last block to handle */
+	struct drvcrypt_buf src; /* Buffer source (Message or Cipher) */
+	struct drvcrypt_buf dst; /* Buffer dest (Message or Cipher) */
+};
+
+/*
+ * Crypto Library Cipher driver operations
+ */
+struct drvcrypt_cipher {
+	/* Allocate of the Software context */
```

Comment:
minor comments:
`Allocate a context` would be fine.
`Free context` for line 51 would e fine also.

As a generic comment, not need for an upper case for Software here and there, or Context, Cipher, ... unless starting a sentence.

Reply:
Change comments to `Allocate/Free context`.
I also have fixed all upper cases